### ISSUE TITLE
Datadog Agent doesn't recognize EC2 Mac instances as running on EC2

### ISSUE THREAD

**Issue by goodspark** [DataDog/datadog-agent#10550]
<details>
<summary>**launchctl**</summary>

```
{
	"StandardOutPath" = "/opt/datadog-agent/logs/launchd.log";
	"LimitLoadToSessionType" = "System";
	"StandardErrorPath" = "/opt/datadog-agent/logs/launchd.log";
	"Label" = "com.datadoghq.agent";
	"OnDemand" = true;
	"LastExitStatus" = 0;
	"PID" = 15977;
	"Program" = "/opt/datadog-agent/bin/agent/agent";
	"ProgramArguments" = (
		"/opt/datadog-agent/bin/agent/agent";
		"run";
	);
};
```
</details>

<details>
<summary>**agent status**</summary>

```
===============
Agent (v7.32.4)
===============

  Status date: 2022-01-14 16:09:37.779 PST / 2022-01-15 00:09:37.779 UTC (1642205377779)
  Agent start: 2022-01-14 16:05:37.888 PST / 2022-01-15 00:05:37.888 UTC (1642205137888)
  Pid: 15977
  Go Version: go1.16.7
  Python Version: 3.8.11
  Build arch: amd64
  Agent flavor: agent
  Check Runners: 6
  Log Level: info

  Paths
  =====
    Config File: /opt/datadog-agent/etc/datadog.yaml
    conf.d: /opt/datadog-agent/etc/conf.d
    checks.d: /opt/datadog-agent/etc/checks.d

  Clocks
  ======
    NTP offset: 83.024ms
    System time: 2022-01-14 16:09:37.779 PST / 2022-01-15 00:09:37.779 UTC (1642205377779)

  Host Info
  =========
    bootTime: 2022-01-14 12:25:56 PST / 2022-01-14 20:25:56 UTC (1642191956000)
    kernelArch: x86_64
    kernelVersion: 20.6.0
    os: darwin
    platform: darwin
    platformFamily: Standalone Workstation
    platformVersion: 11.6.2
    procs: 429
    uptime: 3h39m42s

  Hostnames
  =========
    ec2-hostname: ip-REMOVED.us-east-2.compute.internal
    hostname: buildkite-macos-aws-1
    instance-id: i-0bea064501d6b3d77
    socket-fqdn: buildkite-macos-aws-1
    socket-hostname: buildkite-macos-aws-1
    host tags:
      env:ci
      service:buildkite-agent
    hostname provider: os
    unused hostname providers:
      aws: not retrieving hostname from AWS: the host is not an ECS instance and other providers already retrieve non-default hostnames
      azure: cloud provider is disabled by configuration
      configuration/environment: hostname is empty
      gce: cloud provider is disabled by configuration

  Metadata
  ========
    cloud_provider: AWS
    hostname_source: os

=========
Collector
=========

  Running Checks
  ==============
    
    cpu
    ---
      Instance ID: cpu [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/cpu.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 8, Total: 121
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2022-01-14 16:09:24 PST / 2022-01-15 00:09:24 UTC (1642205364000)
      Last Successful Execution Date : 2022-01-14 16:09:24 PST / 2022-01-15 00:09:24 UTC (1642205364000)
      
    
    disk (4.4.0)
    ------------
      Instance ID: disk:e5dffb8bef24336f [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/disk.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 64, Total: 1,024
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 5ms
      Last Execution Date : 2022-01-14 16:09:31 PST / 2022-01-15 00:09:31 UTC (1642205371000)
      Last Successful Execution Date : 2022-01-14 16:09:31 PST / 2022-01-15 00:09:31 UTC (1642205371000)
      
    
    io
    --
      Instance ID: io [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/io.d/conf.yaml.default
      Total Runs: 15
      Metric Samples: Last Run: 26, Total: 372
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2022-01-14 16:09:23 PST / 2022-01-15 00:09:23 UTC (1642205363000)
      Last Successful Execution Date : 2022-01-14 16:09:23 PST / 2022-01-15 00:09:23 UTC (1642205363000)
      
    
    load
    ----
      Instance ID: load [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/load.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 6, Total: 96
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2022-01-14 16:09:30 PST / 2022-01-15 00:09:30 UTC (1642205370000)
      Last Successful Execution Date : 2022-01-14 16:09:30 PST / 2022-01-15 00:09:30 UTC (1642205370000)
      
    
    memory
    ------
      Instance ID: memory [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/memory.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 9, Total: 144
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2022-01-14 16:09:37 PST / 2022-01-15 00:09:37 UTC (1642205377000)
      Last Successful Execution Date : 2022-01-14 16:09:37 PST / 2022-01-15 00:09:37 UTC (1642205377000)
      
    
    network (2.4.0)
    ---------------
      Instance ID: network:d884b5186b651429 [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/network.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 39, Total: 624
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 17ms
      Last Execution Date : 2022-01-14 16:09:29 PST / 2022-01-15 00:09:29 UTC (1642205369000)
      Last Successful Execution Date : 2022-01-14 16:09:29 PST / 2022-01-15 00:09:29 UTC (1642205369000)
      
    
    ntp
    ---
      Instance ID: ntp:d884b5186b651429 [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/ntp.d/conf.yaml.default
      Total Runs: 1
      Metric Samples: Last Run: 1, Total: 1
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 1
      Average Execution Time : 2ms
      Last Execution Date : 2022-01-14 16:05:39 PST / 2022-01-15 00:05:39 UTC (1642205139000)
      Last Successful Execution Date : 2022-01-14 16:05:39 PST / 2022-01-15 00:05:39 UTC (1642205139000)
      
    
    uptime
    ------
      Instance ID: uptime [OK]
      Configuration Source: file:/opt/datadog-agent/etc/conf.d/uptime.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 1, Total: 16
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2022-01-14 16:09:36 PST / 2022-01-15 00:09:36 UTC (1642205376000)
      Last Successful Execution Date : 2022-01-14 16:09:36 PST / 2022-01-15 00:09:36 UTC (1642205376000)
      
========
JMXFetch
========

  Information
  ==================
  Initialized checks
  ==================
    no checks
    
  Failed checks
  =============
    no checks
    
=========
Forwarder
=========

  Transactions
  ============
    Cluster: 0
    ClusterRole: 0
    ClusterRoleBinding: 0
    CronJob: 0
    DaemonSet: 0
    Deployment: 0
    Dropped: 0
    HighPriorityQueueFull: 0
    Job: 0
    Node: 0
    PersistentVolume: 0
    PersistentVolumeClaim: 0
    Pod: 0
    ReplicaSet: 0
    Requeued: 0
    Retried: 0
    RetryQueueSize: 0
    Role: 0
    RoleBinding: 0
    Service: 0
    ServiceAccount: 0
    StatefulSet: 0

  Transaction Successes
  =====================
    Total number: 32
    Successes By Endpoint:
      check_run_v1: 15
      intake: 2
      series_v1: 15

  API Keys status
  ===============
    API key ending with 5ca1a: API Key valid

==========
Endpoints
==========
  https://app.datadoghq.com - API Key ending with:
      - 5ca1a

==========
Logs Agent
==========


  Logs Agent is not running

=========
APM Agent
=========
  Status: Running
  Pid: 15981
  Uptime: 239 seconds
  Mem alloc: 8,477,880 bytes
  Hostname: buildkite-macos-aws-1
  Receiver: localhost:8126
  Endpoints:
    https://trace.agent.datadoghq.com

  Receiver (previous minute)
  ==========================
    No traces received in the previous minute.
    Default priority sampling rate: 100.0%

  Writer (previous minute)
  ========================
    Traces: 0 payloads, 0 traces, 0 events, 0 bytes
    Stats: 0 payloads, 0 stats buckets, 0 bytes

=========
Aggregator
=========
  Checks Metric Sample: 2,622
  Dogstatsd Metric Sample: 1,849
  Event: 1
  Events Flushed: 1
  Number Of Flushes: 15
  Series Flushed: 3,206
  Service Check: 113
  Service Checks Flushed: 121
=========
DogStatsD
=========
  Event Packets: 0
  Event Parse Errors: 0
  Metric Packets: 1,848
  Metric Parse Errors: 0
  Service Check Packets: 0
  Service Check Parse Errors: 0
  Udp Bytes: 141,466
  Udp Packet Reading Errors: 0
  Udp Packets: 1,304
  Uds Bytes: 0
  Uds Origin Detection Errors: 0
  Uds Packet Reading Errors: 0
  Uds Packets: 0
  Unterminated Metric Errors: 0
```
</details>

<details>
<summary>**config**</summary>

```yaml
api_key: XXXXX

site: datadoghq.com

apm_config:
  enabled: true

tags:
  - service:buildkite-agent

env: ci

cloud_provider_metadata:
  - aws
```
</details>

**Describe what happened:**
Spun up a Mac instance on AWS EC2
Installed the DD agent (7.x)
Viewed the metadata for the machine in the infrastructure host map.

**Describe what you expected:**
Host identified as an EC2 instance and metadata/tags pulled.

![Screenshot from 2022-01-14 16-12-07](https://user-images.githubusercontent.com/29210237/149600632-d8fa90fb-0f8d-417a-aa32-b48c5c511326.png)

Compared to a node running in a GKE cluster with a DD agent running as a DaemonSet:
![Screenshot from 2022-01-14 16-17-22](https://user-images.githubusercontent.com/29210237/149600843-481ff861-ac9f-4ac6-9baf-ab84fe1f8eab.png)

**Steps to reproduce the issue:**
See above

**Additional environment details (Operating System, Cloud provider, etc):**
See above

**Comment by olivielpeau** [comment]
Hi @goodspark, the tags from Google Cloud Platform that you highlighted are collected by Datadog's crawler-based [Google Cloud Platform](https://docs.datadoghq.com/integrations/google_cloud_platform/) integration. The Agent also has the ability to collect these GCP tags, but when it does the tags show up in Datadog as being collected by the Agent.

Similarly, in AWS EC2, the Agent can collect EC2 instance tags when `collect_ec2_tags` is enabled: https://github.com/DataDog/datadog-agent/blob/a60022a9762cc22a4515e952f0fe78153a5c76ac/pkg/config/config_template.yaml#L318-L325
(see the link in the comment for the AWS permission setup required for the Agent to collect tags), and when that's enabled the tags will show up in Datadog as being collected by the Agent. However, you'll want to enable Datadog's crawler-based [EC2 integration](https://docs.datadoghq.com/integrations/amazon_ec2/) if you'd like other EC2-specific metadata and additional metrics to be pulled from AWS EC2.

I'll go ahead and close this issue since I think this answer should help, but feel free to comment here if something's unclear, thanks

**Comment by goodspark** [comment]
Thanks for the tip @olivielpeau ! I missed the EC2 tags setting in the config. I added the AWS Datadog integration and then turned on the EC2 tags in the configs. But I noticed the instance ID is not reported, even though that's readily available in the instance metadata.

Additionally, there's only one Mac instance reported in the host map while we're running a lot more.
![image](https://user-images.githubusercontent.com/29210237/152206129-bbb6124a-7f67-4632-b639-b898a6656e18.png)

How can we:
1. Get instances reporting based on their EC2 instance ID
2. Get the instance ID added to the tags?
